src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,207 @@
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllRequest = "SELECT cars.id, model, manufacturer_id, "
+                + "manufacturers.name AS mark, country FROM cars "
+                + "JOIN manufacturers ON cars.manufacturer_id = "
+                + "manufacturers.id WHERE cars.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllStatement =
+                         connection.prepareStatement(getAllRequest)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithDriverFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't read all cars from DB", e);
+        }
+        cars.forEach(i -> i.setDrivers(getDriversForCar(i.getId())));
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars SET model = ?, manufacturer_id = ? WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateStatement =
+                         connection.prepareStatement(updateRequest)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a car " + car, e);
+        }
+        removeAllRelations(car.getId());
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                     PreparedStatement deleteStatement =
+                             connection.prepareStatement(deleteRequest)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car from DB with id= " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllDriverRequest = "SELECT cars.id, model, manufacturer_id, "
+                + "name AS mark, country "
+                + "FROM cars JOIN manufacturers ON cars.manufacturer_id "
+                + "= manufacturers.id JOIN cars_drivers ON cars.id = "
+                + "cars_drivers.car_id WHERE cars.is_deleted = FALSE AND "
+                + "manufacturers.is_deleted = FALSE AND driver_id = ?;";
```

Comment:
it is better to start each logical part of the SQL query with a new line
Example:
```
SELECT...
FROM...
JOIN....
WHERE...
```